components/dashboard/src/components/IdeOptions.tsx

```diff
@@ -189,8 +209,9 @@ export const IdeOptionsModifyModal = ({
 };
 
 interface IdeOptionSwitchProps {
-    ideOption: IDEOption & { id: string };
+    ideOption: AllowedWorkspaceEditor;
     ideVersions: string[] | undefined;
+    hidePinEditorInputs?: boolean;
```

Comment:
maybe not needed, it should be enogh if `ideVersions` is undefined

Reply:
It's different, we may hide pin button but show the pinned version

Reply:
yep it will work just pass `ideVersions` as undefined and if `pinnedIdeVersion` is not undefined rendet the correct version

Reply:
Addressed https://github.com/gitpod-io/gitpod/pull/19564/commits/d9c7e8675acd0fe482af3ec989b1a9b0ad949ec8